src/service.js

```diff
@@ -101,6 +101,10 @@ export function getExistingServiceInEmbedScope(win, id, opt_fallbackToTopWin) {
 /**
  * Returns a service with the given id. Assumes that it has been constructed
  * already.
+ *
+ * Unlike most service getters, passing `Node` is necessary for some FIE-scope
+ * services since sometimes we only have the FIE Document for context.
+ *
  * @param {!Node|!./service/ampdoc-impl.AmpDoc} nodeOrDoc
```

Comment:
Could we change this to `!Element|!ShadowRoot|!AmpDoc`? That we we fix the core bug here (passing in a `Document`) but still allow `ShadowRoot`s?